src/main/java/com/google/devtools/build/lib/rules/cpp/CppLinkActionBuilder.java

```diff
@@ -1122,7 +1123,8 @@ private CppLinkAction buildLinkAction(
         linkCommandLine,
         linkActionConstruction.getConfig().getActionEnvironment(),
         toolchainEnv,
-        ImmutableMap.copyOf(executionInfo));
+        ImmutableMap.copyOf(executionInfo),
+        PathMappers.getOutputPathsMode(linkActionConstruction.getConfig()));
```

Comment:
This is not reflected in src/main/starlark/builtins_bzl/common/cc/link/finalize_link_action.bzl

(it triggers IFTTT in the file)

Reply:
That's expected, `finalize_link_action.bzl` goes through `ctx.actions.run`, which automatically retains the `OutputPathsMode` in `SpawnAction`. This doesn't require manual code.